C-Plus-Plus/other/First_Missing_Positive_Number.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+Given an array of integers, find the first missing positive integer in linear time and constant space. In other words, 
+find the lowest positive integer that does not exist in the array. The array can contain duplicates and negative numbers as well.
+
+For example, the input [3, 4, -1, 1] should give 2. The input [1, 2, 0] should give 3.
+
+You can modify the input array in-place.
+
+*/
+
+#include<bits/stdc++.h>
+
+using namespace std;
+
+// finding the first missing positive number
+int firstMissingNumber(int arr[], int n)
+{
+    int index, temp;
+    // traversing the array
+    for (index = 0; index < n; index++)
+    {
+        // We will ignore the negative numbers and the numbers greater than the array size
+        // Also, check if the elements are in correct position, if they is not then do swap
+        // operation
+        while (arr[index] != arr[arr[index] - 1] && arr[index] >= 1 && arr[index] <= n)
+        {
+            // Swap operation
+            temp = arr[arr[index] - 1];
+            arr[arr[index] - 1] = arr[index];
+            arr[index] = temp;
+        }
+    }
+    // traversing the array one more time
+    for (index = 0; index < n; index++)
+    {
+        // checking if all the elements are in the correct position
+        // The one which is not in the correct position is the first 
+        // missing positive number
+        if ((index + 1) != arr[index])
+            return index + 1;
+    }
+    return n + 1;
+}
+
+// DRIVER FUNCTION
+int main()
+{
+    // size of the array
+    int n;
```

Comment:
Display suitable message before & after of n
**NOTE:** It is different from Commenting, you have to print it.

Reply:
Could you please elaborate on that @siddharth25pandey ? I don't quite understand.

Reply:
@smv1999 For your better understanding, I am providing you an example
**cout<<Enter the value of n upto which we have to find the first positive missing number**
Similarly others as mentioned.

Reply:
Okay got it! I have made the changes again! Please check if everything is alright!